modelci/controller/controller.py

```diff
@@ -1,47 +1,64 @@
 from modelci.hub.profiler import Profiler
 from modelci.monitor.gpu_node_exporter import GPUNodeExporter
 
-def profiler_callback(model_info, profiler=Profiler, device_util_thd=0, device_memory_thd=0, period=60):
+from threading import Thread
+import time
+
+
+class UtilMonitor(Thread):
+    """
+    Monitor for GPU Utilization
+    """
+
+    def __init__(self, profiler: Profiler, delay, util_level, memory_level):
+        """
+        Init the GPU Utilization Monitor Thread.
+
+        :param delay: time period to get the information.
+        :param util_level: The utilization level that trigger profiling.
+        :param memory_level: The memory usage level that trigger profiling.
+        :param profiler: The instance of model profiler.
+        """
+        super(UtilMonitor, self).__init__()
+        self.stopped = False
+        self.delay = delay
+        self.memory_level = memory_level
+        self.util_level = util_level
+        self.profiler = profiler
+
+        if self.exporter is None:
+            raise TypeError(
+                'Failed to get GPU relative information from node exporter, please make sure the service has started.')
+
+    def run(self):
+        while not self.stopped:
+            exporter = GPUNodeExporter()
+            available_device = exporter.get_idle_gpu(util_level=self.util_level,
+                                                     memory_level=self.memory_level)
+            if available_device is not None:
+                profiler.auto_diagnose(available_devices=available_device,
+                                       batch_list=[8])  # default profiling batch size 8
+            time.sleep(self.delay)
+
+    def stop(self):
+        self.stopped = True
+
+
```

Comment:
https://github.com/anderskm/gputil

better use this toolkit to get the device type here

later we can move it to the exporter class